### ISSUE TITLE
Buffer detection

### ISSUE THREAD

**Issue by achingbrain** [rvagg/cborg#7]
I'm trying to build this module with webpack in [js-ipfs/examples/browser-webpack](https://github.com/ipfs/js-ipfs/tree/master/examples/browser-webpack) and the Buffer detection doesn't work:

```
ERROR in ../../node_modules/cborg/esm/lib/byte-utils.js 1:32-39
Module not found: Error: Can't resolve 'process/browser' in '/Users/alex/Documents/Workspaces/ipfs/js-ipfs/node_modules/cborg/esm/lib'
Did you mean 'browser.js'?
BREAKING CHANGE: The request 'process/browser' failed to resolve only because it was resolved as fully specified
(probably because the origin is a '*.mjs' file or a '*.js' file where the package.json contains '"type": "module"').
The extension in the request is mandatory for it to be fully specified.
Add the extension to the request.
 @ ../../node_modules/cborg/esm/lib/encode.js 9:0-40 192:13-18
 @ ../../node_modules/cborg/esm/cborg.js 1:0-41 7:0-12:2
 @ ../../packages/ipfs-core/src/components/pin/pin-manager.js
 @ ../../packages/ipfs-core/src/components/index.js 93:20-48
 @ ../../packages/ipfs-core/src/index.js 27:4-27
 @ ../../packages/ipfs/src/index.js 12:35-55
 @ ./src/components/app.js 11:11-26
 @ ./src/components/index.js 7:10-26
```

It's good to use buffer because `Buffer.allocUnsafe(size)` is significantly faster than `new Uint8Array(size)`, but it's bad to use node globals.

Solutions:

1. Use the [process](https://www.npmjs.com/package/process) module:

```js
import process from 'process'

export const useBuffer = !process.browser && global.Buffer && typeof global.Buffer.isBuffer === 'function';
```

2. Use the [buffer](https://www.npmjs.com/package/buffer) module:

```js
import { Buffer } from 'buffer'

export const useBuffer = true;

// replace global.Buffer with Buffer
```

**Comment by rvagg** [comment]
What I don't understand about this is that I test this, in the package scripts and in GitHub Actions with Webpack 4 with standard polyfill settings, and I had a user contribute support for 5 without any polyfills. It's not supposed to break with Webpack of all the bundlers! I've gone to a lot of effort here to avoid triggering pulling in the polyfills at all for Webpack 4 even when they're available.

**Comment by achingbrain** [comment]
A fix attempt was made in #11 but this is still broken under certain conditions, notably bundling with [parcel](https://www.npmjs.com/package/parcel).

```
cborg.js_+_16_modules:134 Uncaught ReferenceError: global is not defined
    at eval (cborg.js_+_16_modules:134)
    at Module.19603 (bundle.js:1)
    at __webpack_require__ (bundle.js:1)
    at eval (pin-manager.js:11)
    at Object.78773 (bundle.js:1)
    at __webpack_require__ (bundle.js:1)
    at eval (add-all.js:5)
    at Object.23164 (bundle.js:1)
    at __webpack_require__ (bundle.js:1)
    at eval (index.js:4)
```

**Comment by achingbrain** [comment]
Swapping out `global` for `globalThis` is enough to fix this.
